### PR TITLE
Partially revert 'Remove N+1 in sync'

### PR DESCRIPTION
**Pull request by crisptrutski**
Refences:
- https://github.com/metabase/metabase/issues/38509
- https://github.com/metabase/metabase/issues/38499

### Description

This PR reverts some components of https://github.com/metabase/metabase/pull/37439/

1. We rollback the breaking changes `metabase.driver.sql-jdbc.sync.interface/active-tables` which added a new argument. We will need to maintain backwards compatibility for 3 releases, so will need to introduce a new multi-method instead.
2. We remove the select privileges check from those methods, which required the breaking change, and which we believe is responsible for the two issues linked above.
3. We disable `:table-privileges`for Redshift for now, but leave the queries.

### CODE REVIEW

File: src/metabase/sync/sync_metadata/sync_table_privileges.clj [hunk 1]
```diff
@@ -15,8 +15,7 @@
   [database :- (mi/InstanceOf :model/Database)]
   (let [driver (driver.u/database->driver database)]
     (when (and (not= :redshift driver)
```
Comment (calherries) [inline src/metabase/sync/sync_metadata/sync_table_privileges.clj:17]: You can remove the redshift exclusion now and the comment below
```suggestion
    (when (driver/database-supports? driver :table-privileges database))
```

Comment (calherries) [inline src/metabase/sync/sync_metadata/sync_table_privileges.clj:17]: You can remove the redshift exclusion now and the comment below, so this is just
```clojure
    (when (driver/database-supports? driver :table-privileges database) ...)
```

File: src/metabase/driver/sql_jdbc/sync/interface.clj [hunk 2]
```diff
@@ -9,10 +9,11 @@
 
   Two different implementations are provided in this namespace: `fast-active-tables` (the default), and
   `post-filtered-active-tables`. You should be fine using the default, but refer to the documentation for those
-  functions for more details on the differences."
+  functions for more details on the differences.
+
+  `metabase` is an instance of `DatabaseMetaData`."
```
Comment (calherries) [inline src/metabase/driver/sql_jdbc/sync/interface.clj:14]: I think "metabase" was meant to be "database", but it's gone now anyway
```suggestion
  functions for more details on the differences."
```

**Comment by qnkhuat** [comment]
Are we sure we want to revert this for PG and redshift too? see my [comment](https://metaboat.slack.com/archives/C010L1Z4F9S/p1707393770504909?thread_ts=1707313038.155569&cid=C010L1Z4F9S)

File: src/metabase/sync/sync_metadata/sync_table_privileges.clj (hunk 1 above)
Comment (crisptrutski) [reply src/metabase/sync/sync_metadata/sync_table_privileges.clj:17]: I intentionally updated the comment and left this so we didn't unintentionally enabled it again if we roll the other changes forward again.

**Comment by crisptrutski** [comment]
We have decided against this approach, as it would revert much needed Redshift sync optimization, and we are not aware of any defects in its table-privileges implementation.

**Pull request closed without merging**